src/Matmar10/Bundle/MoneyBundle/Service/CurrencyManager.php

```diff
@@ -99,13 +100,13 @@ protected function searchCurrency($currencyCodeOrCountryCode)
         // search first in the on-the-fly added currencies
         $currency = $this->searchAddedCurrencies($currencyCodeOrCountryCode);
         if(false !== $currency) {
-            return $currency;
+            return $this->addedCurrencies[$currencyCodeOrCountryCode] = $currency; //add to "addedCurrencies" will speed up the next search
```

Comment:
realy ? there is no need to seperate this? just "do it in one line" action style ;)